File: typescript/src/data_models/user_task/user_task_config.ts
import {UserTaskFormField} from './user_task_form_field';

/**
 * Contains information about a UserTasks configuration.
 */
export class UserTaskConfig {
  /**
   * A list of accessible form fields for the UserTask.
   */
  public formFields: Array<UserTaskFormField> = [];
  /**
   * The perferred type of control for the form field.
   */
  public preferredControl?: string;
  /**
   * Optional: A description of this UserTask.
   */
  public description?: string;
  /**
   * Optional: A message to show when the UserTask has been finished.
   */
  public finishedMessage?: string;
}

File: dotnet/src/data_models/user_task/UserTaskConfig.cs
namespace ProcessEngine.ConsumerAPI.Contracts.DataModel
{
    using System.Collections.Generic;

    /// <summary>
    /// Contains information about a UserTasks configuration.
    /// </summary>
    public class UserTaskConfig
    {
        /// <summary>
        /// A list of accessible FormFields for the UserTask.
        /// </summary>
        public IEnumerable<UserTaskFormField> FormFields { get; set; }

        /// <summary>
        /// The perferred type of control for the FormField.
        /// </summary>
        public string PreferredControl { get; set; }

        /// <summary>
        /// Optional: A description of this UserTask.
        /// </summary>
        public string Description { get; set; }

        /// <summary>
        /// Optional: A message to show when the UserTask has been finished.
        /// </summary>
        public string FinishedMessage { get; set; }
    }
}
